x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/PropagateEvalFoldables.java

```diff
@@ -32,6 +33,10 @@ public LogicalPlan apply(LogicalPlan plan) {
         // collect aliases bottom-up
         plan.forEachExpressionUp(Alias.class, a -> {
             var c = a.child();
+            if (c instanceof Categorize) {
+                // Categorize cannot be folded
+                return;
+            }
```

Comment:
Jesus Christ, this logic makes no sense, my bad.
I'll just make Categorize non-foldable, and add that comment there :hehe:

Reply:
Just double checked: The condition isn't needed because this rule doesn't replace anything in aggregations. Only filters and evals. So the grouping remains untouched